google-http-client/src/main/java/com/google/api/client/http/HttpRequest.java

```diff
@@ -1159,7 +1159,7 @@ public boolean handleRedirect(int statusCode, HttpHeaders responseHeaders) {
         && HttpStatusCodes.isRedirect(statusCode)
         && redirectLocation != null) {
       // resolve the redirect location relative to the current location
-      setUrl(new GenericUrl(url.toURL(redirectLocation)));
+      setUrl(new GenericUrl(url.toURL(redirectLocation), true));
```

Comment:
I feel like we should preserve the current behavior, and the verbatim behavior should be turned on manually with some option.

Reply:
Makes sense!
I'll try to find time and add it.

How do you test it with Jib? Tried to build Jib with a snapshot build of google-http-client and run into tons of Guava related issues....

Reply:
> How do you test it with Jib?

I'll actually have to start working on this. I'll share something when I make it work. This will take some time.

Reply:
I've pushed up the branch [`upgrade-google-http-client`](https://github.com/GoogleContainerTools/jib/tree/upgrade-google-http-client) that is set up to use `google-http-client-1.33.1-SNAPSHOT` and sets `setUseRawDirectUrls(true)`. Just do

```
google-http-client$ mvn clean install
google-http-client$ cd $JIB_REPO
jib$ git fetch
jib$ git checkout upgrade-google-http-client
jib$ ./gradlew clean verify install
```

And it will install Jib 1.7.1-SNAPSHOT locally. If you have any trouble using 1.7.1-SNAPSHOT, let me know.

And please test it on both of these cases (https://github.com/GoogleContainerTools/jib/issues/1986#issuecomment-547610104), where one is encoding-related and the other is decoding-related.

- Encoding-related
```
from: exp=1572285389~hmac=f0a387f...     --> OpenShift says OK
to  : exp%3D1572285389~hmac%3Df0a387f... --> confuses OpenShift
```
- Decoding-related
```
from: Signature=2wYOD0a%2BDAkK%2F9lQ... --> Quay says OK
to  : Signature=2wYOD0a%2BDAkK/9lQ...   --> confuses Quay
```